pkg/webhook/pod/validating/pod_unavailable_budget.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2021 The Kruise Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validating
+
+import (
+	"context"
+
+	"github.com/openkruise/kruise/pkg/control/pubcontrol"
+
+	admissionv1beta1 "k8s.io/api/admission/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apiserver/pkg/util/dryrun"
+	"k8s.io/klog"
+	"k8s.io/kubernetes/pkg/apis/policy"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+func (p *PodCreateHandler) podUnavailableBudgetValidatingPod(ctx context.Context, req admission.Request) (allowed bool, reason string, err error) {
+	allowed, newPod, oldPod, dryRun, err := p.shouldIgnore(ctx, req)
```

Comment:
The name `shouldIgnore` is confusing. If it is named this way, should line 41 - 47 be moved into the `shouldIgnore` method?